src/core/dag/StochasticNode.h

```diff
@@ -463,20 +459,17 @@ double RevBayesCore::StochasticNode<valueType>::getLnProbability( void )
         {
             lnProb = 0.0;
         }
-        
-        // reset flag
-        needs_probability_recalculation = false;
     }
-    
-    return lnProb;
+
+    return lnProb.value();
 }
 
 
 template<class valueType>
 double RevBayesCore::StochasticNode<valueType>::getLnProbabilityRatio( void )
 {
     
-    return getLnProbability() - stored_ln_prob;
+    return getLnProbability() - stored_ln_prob.value().value();
```

Comment:
It throws an exception if we do not, or if we do and it is not calculated.  Perhaps surprisingly, that exception was not thrown.

Are you thinking that if we do not have a stored value, then we should return zero?

That makes sense... I did not do this because the function is only used in two places, but I could add it.

Reply:
yes I'm thinking we should return zero if there is no stored value - otherwise we depend on the calling code to only call this for the "correct" nodes 
I'm less clear on what we should do if the value exists but has not been calculated - I guess throwing an exception is fine in that case, but maybe it should be documented

Reply:
Sure, that make sense.  I'll return zero first, and possibly eliminate the function later.